src/main/scala/linting/rule/LintAmbiguousModuleNames.scala

```diff
@@ -0,0 +1,331 @@
+  }
+
+  def removePortInfo(port: Port): Port = {
+    port.copy(info = NoInfo)
+  }
+
+  def removeModuleInfo(mod: Module): Module = {
+    mod.copy(
+      ports = mod.ports.map(removePortInfo(_)),
+      body = removeStatementInfo(mod.body),
+      info = NoInfo)
+  }
+}
+
+/** Specifies a naming strategy to use for a module and the modules that it collides with
+  */
+case class NamingStrategyAnnotation(
+  strategy: NamingStrategy,
+  target: IsModule
+) extends SingleTargetAnnotation[IsModule] with HasSerializationHints {
+  def typeHints: Seq[Class[_]] = Seq(
+    ExactNamingStrategy.getClass,
+    PortStructureNamingStrategy.getClass,
+    ContentStructureNamingStrategy.getClass,
+    ContentNamingStrategy.getClass
+  )
+
+  def duplicate(newTarget: IsModule): NamingStrategyAnnotation = {
+    this.copy(target = newTarget)
+  }
+}
+
+/** Specifies a desired name for a module
+  */
+case class ModuleNameAnnotation(
+  desiredName: String,
+  target: IsModule
+) extends SingleTargetAnnotation[IsModule] {
+  def duplicate(newTarget: IsModule): ModuleNameAnnotation = {
+    this.copy(target = newTarget)
+  }
+}
+
+/** An example of how to customize the names of certain modules using an Aspect
+  */
+case object StabilizeNamesAspect extends Aspect[RawModule] {
+  def toAnnotation(top: RawModule): AnnotationSeq = {
+    Select.collectDeep(top) {
+      // annotating all Queues with a more descriptive desired name
+      case m: Queue[_] => Seq(new ModuleNameAnnotation(s"Queue_${m.genType.getClass.getSimpleName}_entries_${m.entries}", m.toTarget))
+      case m: TLMonitor => Seq(new ModuleNameAnnotation(m.desiredName, m.toTarget))
+
+      // annotating specific instances
+      case th: TestHarness =>
+        val core = th.ldut.rocketTiles.head.module.core
+        Seq(
+          ModuleNameAnnotation("Rocket_Core_0", core.toTarget),
+          NamingStrategyAnnotation(ExactNamingStrategy, core.toTarget),
+        )
+    }.flatten.toSeq
+  }
+}
+
+/** This LintRule checks for module name collisions and optionally renames them to stable hash names
+  *
+  * Module name collisions occur when different [[Module]]s are annotated with
+  * [[ModuleNameAnnotation]]s that have the same desiredName. If one of the
+  * modules is targeted by [[NamingStrategyAnnotation]] then the transform will
+  * attempt to rename modules according to the naming strategy. If the
+  * conflicting modules cannot be disambiguated by the naming strategy then a
+  * [[LintViolation]] is emitted. [[ExactNamingStrategy]] can be used to
+  * enforce that there are not colliding module names.
+  *
+  * If no naming strategy is specified then a sequence of increasingly unstable
+  * [[NamingStrategy]]s is tried until all modules can be disamgiguated.
+  */
+final class LintAmbiguousModuleNames extends LintRule {
+  val recommendedFix: String = "override desiredName based on module parameters"
+
+  val lintName: String = "ambiguous-module-names"
```

Comment:
should it be `ambiguous-module-names` or `conflicting-module-names`?

Reply:
I think conflicting-module-names is better